product/opni/models/Document.ts

```diff
@@ -0,0 +1,28 @@
+import { Resource } from './Resource';
+
+export interface ConfigDocumentWithSchema {
+  json: string;
+  yaml: string;
+  schema: string;
+}
+
+export interface GatewayConfig {
+  documents: ConfigDocumentWithSchema[];
+}
+
+export class Document extends Resource {
```

Comment:
Is there something else we could call this other than document? There's a global variable in browsers called document which could result in a naming collision if someone creates a variable named document or accidentally tries to access a document variable that wasn't defined.